src/main/kotlin/org/openmbee/flexo/sysmlv2/apis/CommitApi.kt

```diff
@@ -349,53 +388,116 @@ fun Route.CommitApi() {
+                values.add(elementNode.stringify())
+
+//                // generate the WHERE clause
+//                wheres.add("""
+//                    ${elementNode.stringify()} ?p_$index ?o_$index .
+//
+//                    optional {
+//                        ?incoming ?incoming_relation_p ${elementNode.stringify()} ;
+//                            ?incoming_order_p ?incoming_order_o .
+//
+//                        filter(
+//                            str(?incoming_order_p) = concat(
+//                                "${SYSMLV2.ANNOTATION_JSON}:",
+//                                strAfter(
+//                                    str(?incoming_relation_p),
+//                                    "${SYSMLV2.PROPERTY}:"
+//                                )
+//                            )
+//                        )
+//                    }
+//                """.trimIndent())
             }
         }
 
-        // forward failures to client
-        if(flexoResponseLock.isFailure()) {
-            return@post forward(flexoResponseLock)
-        }
+//        // first, lock the given commit
+//        val flexoResponseLock = flexoRequestPost {
+//            orgPath("/repos/$projectId/locks")
+//
+//            turtle {
+//                """
+//                    <> mms:commit mor-commit:${"" /*commit.previousCommit*/} .
+//                """.trimIndent()
+//            }
+//        }
+//
+//        // forward failures to client
+//        if(flexoResponseLock.isFailure()) {
+//            return@post forward(flexoResponseLock)
+//        }
+//
+//        // extract lock ID from response model
+//        val lockId = flexoResponseLock.parseLdp {
+//            primary[MMS.id].literal()!!
+//        }
 
-        // extract lock ID from response model
-        val lockId = flexoResponseLock.parseLdp {
-            primary[MMS.id].literal()!!
-        }
-*/
-        val body = """
-                   delete {
-                        ${deletes.joinToString("\n")}
+        val lockId = "none"
+
+        // submit POST request to commit model
+        val flexoResponseUpdate = flexoRequestPost {
+            orgPath("/repos/$projectId/locks/$lockId/update")
+
+            // construct body payload
+            sparqlUpdate {
+                """
+                    delete {
+                        ?element_n ?element_p ?element_o .
+
+                        ?incoming ?incoming_p ?element_n ;
+                                ?incoming_order_p ?incoming_order_o .
                     }
                     insert {
                         ${inserts.joinToString("\n")}
                     }
                     where {
-                        ${wheres.joinToString("\n")}
                         optional {
-                            <urn:this> <urn:always> <urn:matches> .
+                            ?element_n ?element_p ?element_o .
+                        }
+                        
+                        optional {
```

Comment:
i thought we're not doing the delete cascade stuff? and the list ordering are not in triples anymore

Reply:
revised to only delete incoming order triples when element is being deleted (no cascading beyond 1-degree)